delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CharStreams;
+import io.cdap.cdap.api.ServiceDiscoverer;
+import io.cdap.cdap.api.retry.RetryableException;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.delta.app.service.AssessmentService;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.FailsafeExecutor;
+import net.jodah.failsafe.RetryPolicy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import javax.annotation.Nullable;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Class to handle Api calls to System Service
+ */
+public class AssessmentServiceClient {
+
+  ServiceDiscoverer serviceDiscoverer;
+
+  public AssessmentServiceClient(ServiceDiscoverer context) {
+    this.serviceDiscoverer = context;
+  }
+
+  public String retryableApiCall(String connectionUrl, String method, @Nullable String requestBody) {
+
+    RetryPolicy retryPolicy = new RetryPolicy()
+      .withMaxAttempts(5)
+      .withMaxDuration(Duration.of(10, ChronoUnit.SECONDS))
+      .withDelay(Duration.of(200, ChronoUnit.MILLIS))
+      .withJitter(0.20D);
+
+    FailsafeExecutor<String> failsafe = Failsafe.with(retryPolicy);
+
+    return failsafe.get(() -> {
+      if (method.equals(HttpMethod.GET)) {
+        return executeGetCall(getHttpUrlConnection(connectionUrl));
+      } else if (method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT)) {
+        return executeCallWithBody(getHttpUrlConnection(connectionUrl), method, requestBody);
+      }
+      throw new IllegalStateException("The given Http Method Doesn't match with available methods");
+    });
+
+  }
+
+  private String executeGetCall(HttpURLConnection urlConn) throws IOException {
```

Comment:
as commented elsewhere, this should not be reading the response as a string and then converting the string into an object. Instead, it should convert directly to the object with the Reader.

There also needs to be a separate method thats used when the response is bytes and not json.